cou/apps/base.py

```diff
@@ -698,9 +698,11 @@ def _get_upgrade_charm_steps(self, target: OpenStackRelease) -> list[UpgradeStep
             ]
 
         raise ApplicationError(
-            f"The '{self.name}' application is using channel '{self.channel}'. Channels supported "
-            f"during this transition: '{self.expected_current_channel(target)}', "
-            f"'{self.target_channel(target)}'. Manual intervention is required."
+            f"The '{self.name}' application is using an unexpected channel: '{self.channel}'. "
+            "Channels supported during this upgrade are, "
+            f"before upgrade: '{self.expected_current_channel(target)}', "
```

Comment:
I'm confused about this, because the error message implies that only two channels are supported, but the logic above is more complex (`self.channel_o7k_release <= self.o7k_release or self.multiple_channels`). @gabrielcocenza @rgildein could you explain more about the requirements here?

Reply:
Normally the target channel is N+1 from the current_channel. If it's already in the target, we don't create this steps.

When two or more applications are colocated, when you upgrade app A, the payload of app B is also upgraded. That is why you can have the o7k release of the channel lesser than the payload version and you still need to change the channel to be in the same page. That explains the `self.channel_o7k_release <= self.o7k_release` .

The `multiple_channels` are a corner case of RabbitMQ and HACluster that can have a channel different from what is on [charm-delivery](https://docs.openstack.org/charm-guide/latest/project/charm-delivery.html). In summary you can have Rabbitmq on ussuri using channek 3.8 or channel 3.9. We decided that when we see Rabbitmq on 3.8 we will change to 3.9 so that is why is necessary to add this `self.multiple_channels`

Reply:
Hmm ok, but what's the standard case? Here the method is "Get steps for upgrading the charm", but it seems that it's the corner cases (colocated or `multiple_channels`) that get the upgrade steps; other upgrade cases get the ApplicationError here. :thinking: